lib/scan/wp_xmlrpc/engine.py

```diff
@@ -0,0 +1,205 @@
+import threading
+import string
+import requests
+import random
+import os
+import re
+from core.alert import *
+from core.targets import target_type
+from core.targets import target_to_host
+from core.load_modules import load_file_path
+from lib.socks_resolver.engine import getaddrinfo
+from core._time import now
+from core.log import __log_into_file
+from core._die import __die_failure
+from lib.payload.wordlists.useragents import useragents
+
+def extra_requirements_dict():
+    return {
+        "wp_xmlrpc_scan_ports": [80, 443]
+    }
+
+
+def check(target, port, headers, timeout_sec, log_in_file, language,
+            retries, time_sleep, thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+    time.sleep(time_sleep)
+    try:
+        if socks_proxy is not None:
+            socks_version = socks.SOCKS5 if socks_proxy.startswith(
+                'socks5://') else socks.SOCKS4
+            socks_proxy = socks_proxy.rsplit('://')[1]
+            if '@' in socks_proxy:
+                socks_username = socks_proxy.rsplit(':')[0]
+                socks_password = socks_proxy.rsplit(':')[1].rsplit('@')[0]
+                socks.set_default_proxy(socks_version, str(socks_proxy.rsplit('@')[1].rsplit(':')[0]),
+                                        int(socks_proxy.rsplit(':')[-1]), username=socks_username,
+                                        password=socks_password)
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+            else:
+                socks.set_default_proxy(socks_version, str(
+                    socks_proxy.rsplit(':')[0]), int(socks_proxy.rsplit(':')[1]))
+                socket.socket = socks.socksocket
+                socket.getaddrinfo = getaddrinfo
+        n = 0
+        while 1:
+            try:
+                if target_type(target) != "HTTP" and port == 443:
+                    target = 'https://' + target
+                if target_type(target) != "HTTP" and port == 80:
+                    target = 'http://' + target
+                target = target + '/xmlrpc.php'
+                postdata = '''<?xml version="1.0" encoding="utf-8"?><methodCall><methodName>system.listMethods</methodName><params></params></methodCall>'''
+                r = requests.post(
+                        target, timeout = timeout_sec, headers = headers, data = postdata)
+                if "demo.sayhello" in r.text.lower():
+                    info(messages(language, "target_vulnerable").format(
+                                    target, port, "XMLRPC DOS attacks"))
+                    __log_into_file(thread_tmp_filename, 'w', '0', language)
+                    data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': port, 'TYPE': 'wp_xmlrpc_scan',
+                               'DESCRIPTION': messages(language, "vulnerable").format("XML-RPC DOS attacks!!") , 'TIME': now(), 'CATEGORY': "brute", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd}) + "\n"
```

Comment:
please change to 'vuln'.  It should be 'vuln' not 'brute' - because Nettacker is checking for a vulnerability. Category 'brute' actually performs a brute forcing attack by enumerating users/passwords